src/components/spreadsheet/add-new-spreadsheet/add-new-spreadsheet-dialog.tsx

```diff
@@ -0,0 +1,204 @@
+                const newTableDefinition = {
+                    ...TABLES_DEFINITIONS[equipmentType],
+                    name: newParams[SPREADSHEET_NAME],
+                    index: tablesDefinitionIndexes.size,
+                };
+                dispatch(
+                    updateTableDefinition(
+                        'new' + tablesDefinitionIndexes.size + newParams.equipmentType,
+                        newTableDefinition,
+                        []
+                    )
+                );
+            } else {
+                getSpreadsheetModel(newParams[SPREADSHEET_MODEL][0].id).then(
+                    (selectedModel: { customColumns: any; sheetType: keyof typeof TABLES_DEFINITIONS }) => {
+                        const newTableDefinition = {
+                            ...TABLES_DEFINITIONS[selectedModel.sheetType],
+                            name: newParams[SPREADSHEET_NAME],
+                            index: tablesDefinitionIndexes.size,
+                            columns: [],
+                        };
+                        dispatch(
+                            updateTableDefinition(
+                                'new' + tablesDefinitionIndexes.size + newParams[SPREADSHEET_MODEL][0].name,
+                                newTableDefinition,
+                                selectedModel.customColumns
+                            )
+                        );
+                    }
+                );
+            }
+            open.setFalse();
+        },
+
+        [dispatch, open, selectedOption?.id, tablesDefinitionIndexes.size]
+    );
+
+    useEffect(() => {
+        reset(initialEmptySpreadsheetForm);
+    }, [open.value, reset]);
+
+    const addEmptySpreadsheetConfigContent = (
+        <Grid container spacing={2} direction="column">
+            <Grid item xs>
+                <TextInput
+                    name={SPREADSHEET_NAME}
+                    label={'spreadsheet/create_new_spreadsheet/spreadsheet_name'}
+                    formProps={{ autoFocus: true }}
+                />
+            </Grid>
+            <Grid item xs>
+                <SelectInput
+                    options={Object.values(EQUIPMENT_TYPES).map((equipmentType) => {
+                        return { id: equipmentType, label: equipmentType };
+                    })}
+                    name={EQUIPMENT_TYPE_FIELD}
+                    label={'spreadsheet/create_new_spreadsheet/equipment_type'}
+                    size={'small'}
+                />
+            </Grid>
+        </Grid>
+    );
+
+    const addSpreadsheetFromModelContent = (
+        <Grid container spacing={2} direction="column">
+            <Grid item xs>
+                <TextInput
+                    name={SPREADSHEET_NAME}
+                    label={'spreadsheet/create_new_spreadsheet/spreadsheet_name'}
+                    formProps={{ autoFocus: true }}
```

Comment:
If the template is selected, the Name field does not take the name of the template